test/support/src/vfs_helpers.h

```diff
@@ -749,6 +811,57 @@ struct TemporaryDirectoryFixture {
   const std::vector<std::unique_ptr<SupportedFs>> supported_filesystems_;
 };
 
+/*
+ * Class to instantiate when setting up a test case to run for all VFSs and
+ * REST
+ *
+ * Example usage: In the beginning of the test define a VFSTestSetup variable
+ * and then just use the Context and the rest of the resources it encapsulates
+ * in the test. The resources will be automatically get freed when the variable
+ * will get out of scope.
+ *
+ * {
+ * tiledb::test::VFSTestSetup vfs_test_setup{"foo_array"};
+ * auto ctx = vfs_test_setup.ctx;
+ * Array array(ctx, array_uri, TILEDB_WRITE);
+ * ...
+ * } // ctx context is destroyed and VFS directories removed
+ *
+ */
+struct VFSTestSetup {
+  VFSTestSetup() = delete;
+
+  VFSTestSetup(std::string array_name, tiledb_config_t* config = nullptr)
+      : fs_vec(vfs_test_get_fs_vec())
+      , ctx_c{nullptr}
+      , vfs_c{nullptr}
+      , cfg_c{config} {
+    vfs_test_init(fs_vec, &ctx_c, &vfs_c, cfg_c).ok();
+    ctx = Context(ctx_c, false);
```

Comment:
If my understanding of C++'s initialization rules is correct, the declaration of `ctx` at the bottom of the class already creates a context, and this line destroys it. Maybe the following function instead of the `ctx` field will prevent this:

```c++
  Context ctx() {
    return Context(ctx_c, false);
  }
```